api/src/main.rs

```diff
@@ -546,23 +559,28 @@ async fn main() -> color_eyre::Result<()> {
     tracing::info!("Running SQLx migrations");
     sqlx::migrate!("sqlx/migrations").run(&db).await?;
 
+    let json = ApiDoc::openapi();
     let api = Router::new()
-        .route("/search/list/:name", get(search_list))
-        .route("/search/account/:name", get(search_account))
-        .route("/history/:id", get(history_search))
+        .route("/search/list/{name}", get(search_list))
+        .route("/search/account/{name}", get(search_account))
+        .route("/history/{id}", get(history_search))
         .nest("/list", list::router())
+        .nest("/account", auth::router())
         .nest("/share", share::router())
-        .nest("/account", account::router())
-        .nest("/pantry", pantry::router());
+        .nest("/pantry", pantry::router())
+        .route("/openapi.json", get(|| async { Json(json) }));
 
     #[cfg(feature = "frontend")]
     let frontend = config.frontend.clone();
 
-    let app = Router::new()
-        .merge(SwaggerUi::new("/swagger-ui").url("/api-doc/openapi.json", ApiDoc::openapi()))
+    let app: Router = Router::new()
+        // .merge(Router::<()>::from(SwaggerUi::new("/swagger-ui").url("/api-doc/openapi.json", ApiDoc::openapi()))
```

Comment:
Commented code

Reply:
This is so if someone wishes to actually have SwaggerUi installed they can. Currently this requires using a Patched version of utoipa since they don't support axum 0.8.0.